### PR TITLE
Ordering Gate Batch Support

### PR DESCRIPTION
**Pull request by Akvinikym**
### Description of the Change

Support for transaction batches was added to ordering gate, so that they are propagated through the pipeline, for now ending in ordering endpoint, from where in future PR it will be handled by ordering service.
Tests are not added, because there is no sense it testing propagation functions without implemented ordering service support.

### Benefits

More batch support.

### Possible Drawbacks 

None

### CODE REVIEW

File: irohad/ordering/impl/ordering_gate_transport_grpc.cpp [hunk 1]
```diff
@@ -73,6 +74,23 @@ void OrderingGateTransportGrpc::propagateTransaction(
   call->response_reader->Finish(&call->reply, &call->status, call);
 }
 
+void OrderingGateTransportGrpc::propagateBatch(
+    const shared_model::interface::TransactionBatch &batch) {
+  log_->info("Propagate transaction batch (on transport)");
+  auto call = new AsyncClientCall;
+
+  iroha::protocol::TxList batch_transport;
+  for (const auto tx : batch.transactions()) {
```
[the remark this answers is not in the document]
Comment (kamilsa) [reply irohad/ordering/impl/ordering_gate_transport_grpc.cpp]: No need, it is shared_ptr

File: irohad/ordering/impl/ordering_gate_impl.cpp [hunk 2]
```diff
@@ -43,14 +43,22 @@ namespace iroha {
...
       log_->info("propagate tx, account_id: {}",
                  " account_id: " + transaction->creatorAccountId());
 
       transport_->propagateTransaction(transaction);
     }
 
+    void OrderingGateImpl::propagateBatch(
+        const shared_model::interface::TransactionBatch &batch) const {
+      log_->info("propagate batch, account_id: {}",
+                 batch.transactions().front()->creatorAccountId());
```
[the remark this answers is not in the document]
Comment (kamilsa) [reply irohad/ordering/impl/ordering_gate_impl.cpp]: +1. Write smth like `if (batch.empty()) return;`

File: schema/ordering.proto [hunk 3]
```diff
@@ -3,6 +3,7 @@ package iroha.ordering.proto;
 
 import "transaction.proto";
 import "proposal.proto";
+import "endpoint.proto";
```
[the remark this answers is not in the document]
Comment (kamilsa) [reply schema/ordering.proto]: For tx list, I guess

File: schema/ordering.proto [hunk 4]
```diff
@@ -11,4 +12,5 @@ service OrderingGateTransportGrpc {
 
 service OrderingServiceTransportGrpc {
   rpc onTransaction (iroha.protocol.Transaction) returns (google.protobuf.Empty);
+  rpc onBatch (iroha.protocol.TxList) returns (google.protobuf.Empty);
```
Comment (kamilsa) [inline schema/ordering.proto]: Why it is here btw? I thought onBatch will be implemented in a separate PR

File: irohad/ordering/impl/ordering_gate_impl.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (l4l) [reply irohad/ordering/impl/ordering_gate_impl.cpp]: silent return probably isn't a desired behavior

File: schema/ordering.proto (hunk 4 above)
Comment (Akvinikym) [reply schema/ordering.proto]: Because otherwise gRPC call method will not be generated. It's needed in ordering_gate_transport_grpc.cpp, line 89
